### PR TITLE
Fix for race condition in partition state clean/dirty tracking

### PR DESCRIPTION
**Pull request by rkolesnev**
Fix for race condition in partition state clean/dirty tracking - see #664 
### Checklist

- [X] Documentation (if applicable)
- [X] Changelog

### CODE REVIEW

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/PartitionState.java [hunk 1]
```diff
@@ -382,9 +396,14 @@ public boolean isRemoved() {
...
-                of(createOffsetAndMetadata()) :
-                empty();
+        if (isDirty()) {
+            // setting the flag so that any subsequent offset completed while commit is being performed could mark state as dirty
+            // and retain the dirty state on commit completion.
+            stateChangedSinceCommitStart = false;
+            return of(createOffsetAndMetadata());
+        } else {
+            return empty();
+        }
```
Comment (eddyv) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/PartitionState.java:406]: nitpick
```suggestion
        }
        return empty();
```

**Review by eddyv — APPROVED** [review]
small nitpick

**Review by eddyv — CHANGES_REQUESTED** [review]
```
09:12:18  [ERROR] /home/jenkins/workspace/builder_parallel-consumer_PR-666/parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/state/PartitionStateCommittedOffsetTest.java:[185,137] cannot find symbol
09:12:18  [ERROR]   symbol:   method of(long,long)
09:12:18  [ERROR]   location: interface java.util.List
```
Missing something 😅

**Comment by rkolesnev** [comment]
> ```
> 09:12:18  [ERROR] /home/jenkins/workspace/builder_parallel-consumer_PR-666/parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/state/PartitionStateCommittedOffsetTest.java:[185,137] cannot find symbol
> 09:12:18  [ERROR]   symbol:   method of(long,long)
> 09:12:18  [ERROR]   location: interface java.util.List
> ```
> 
> Missing something 😅

Ah - local build on newer java... while CI build on java 8 i think - which ofc doesn't have List.of(...) support

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/PartitionState.java (hunk 1 above)
Comment (rkolesnev) [reply parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/PartitionState.java]: done

**Review by eddyv — APPROVED** [review]

**Pull request merged**